### ISSUE TITLE
Decryption Errors

### ISSUE THREAD

**Issue by tkcranny** [backbone-hq/minibone#1]
Hey folks. I've been curious about client-side encryption and have been trying out Minibone.

I've been running into some errors when I call `.decrypt()` on Minibone instances. This call even fails with different error messages between different browsers, albeit from the same calls.

I'm using a client-side SvelteKit application. Creating Minibone instances, generating the payloads, and encrypting data all work fine, it's just the `.decrypt` method that seems to be problematic, presumably because of something lower-level in the WebCrypto API.

I've been able to get it down to the below example. I've also made a rudimentary [StackBlitz example](https://stackblitz.com/edit/sveltejs-kit-template-default-ywsvk1?file=src%2Froutes%2F%2Bpage.js) that replicates it. 

```typescript
const miniboneTest = async (userId: string, secretKey: string, input: string) => {
  const inMb = await Minibone.create();
  const key = await inMb.save(secretKey, ['app.my.domain', userId]);
  const enc = await inMb.encrypt({ input }, key);
  const outMb = await Minibone.load(key, secretKey, ['app.my.domain', userId]);
  const decrypted = await outMb.decrypt(enc); // Crashes here.
};
```

This crashes on the last `.decrypt(call)`.

If the library isn't intended for frontend applications / browsers, or I'm invoking it incorrectly, then I'm all ears. 

Interestingly the error is also different between different browsers, Chrome gives:
```
Uncaught (in promise) TypeError: Failed to execute 'decrypt' on 'SubtleCrypto': AesGcmParams: additionalData: Not a BufferSource.
```

Where Safari and Firefox give:
```
Unhandled Promise Rejection: OperationError: The operation failed for an operation-specific reason.
```

A screenshot of my playground page for this, 
<img width="2055" alt="image" src="https://github.com/backbone-hq/minibone/assets/4208581/b12a0d8d-34ac-48ca-9a1c-14654c6d5341">

**Comment by david-alm** [comment]
Hey @tkcranny. Thanks for filing this issue and taking the time to set up a minimal reproducible example.
You really made debugging this a breeze.

To cut straight to the chase, the immediate fix for your code example is to replace
```ts
- const enc = await inMb.encrypt({ input }, key)
+ const enc = await inMb.encrypt(input)
```

The reason this fails is subtle, but fortunately is expected behaviour. 
That said, Minibone should do a better job here and produce helpful error messages.

**To delve into the details:**
The issue is that the <a href="https://github.com/backbone-hq/minibone/commit/c29c3e0e2c34af01e018431f194f2839d417e561#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R149">encrypt</a> function accepts an <b>optional</b> <code>associatedData</code>. The same associated data <b>must</b> be passed during decryption, because it's used to compute a message authentication code (a form of <it>symmetric</it> digital signature).

`associatedData` is used to add context to verify the integrity of data to protect against server-side tampering, replay attacks or other such nefarious modification of the ciphertext by an attacker. The decision to use it depends on your threat model.

In short, another viable fix here would be something like
```ts
- const enc = await inMb.encrypt({ input }, key)
+ const enc = await inMb.encrypt(input, userId)

- const decrypted = await outMb.decrypt(enc); // Crashes here.
+ const decrypted = await outMb.decrypt(enc, userId)
```

Hopefully this clarifies the intended usage.
I'm going to close this issue out, but feel free to open it if you feel it's still unsolved.

(P.S. Congratulations on filing minibone issue #1)
